diff --git a/xpcom/idl-parser/xpidl.py b/xpcom/idl-parser/xpidl.py
--- a/xpcom/idl-parser/xpidl.py
+++ b/xpcom/idl-parser/xpidl.py
@@ -611,16 +611,19 @@ class ConstMember(object):
             basetype = basetype.realtype
         if not isinstance(basetype, Builtin) or not basetype.maybeConst:
             raise IDLError("const may only be a short or long type, not %s" % self.type, self.location)
 
         self.basetype = basetype
 
     def getValue(self):
         return self.value(self.iface)
+
+    def __str__(self):
+        return "\tconst %s %s = %s\n" % (self.type, self.name, self.getValue())
 
 class Attribute(object):
     kind = 'attribute'
     noscript = False
     notxpcom = False
 
     def __init__(self, type, name, attlist, readonly, location, doccomments):
         self.type = type
